src/VisionModel/resnet.jl

```diff
@@ -0,0 +1,93 @@
+using Flux
+
+function basicblock()
+  dilation = 1;
+  if dilation > 1
+    error("Dilation > 1 not supported in Basic Block");
+  end
+  basic =  Chain(
+      Conv((3, 3), inplanes=>planes, stride=stride, pad=(1,1), dilation=dilation),
+      BatchNorm(planes, λ = relu),
+      Conv((3, 3), planes=>planes, stride=stride, pad=(1,1), dilation=dilation),
+      BatchNorm(planes)
+    );
+  return basic
+end
+
+function bottleneck(planes::Int, stride::Int)
+  stride = 2
+  base_width = 64;
+  groups = 1;
+  width = Int(planes * (base_width / 64.)) * groups;
+  residual =  Chain(
+          Conv((1, 1), inplanes=>width, stride=(2,2)),
+          BatchNorm(width, λ=relu),
+          Conv((3, 3), width=>width, stride=(2,2), pad=(1,1), dilation=dilation),
+          BatchNorm(width, λ=relu),
+          Conv((1, 1), width=>planes * expansion, stride=(2,2)),
+          BatchNorm(planes * expansion, λ=relu)
+          );
+  return residual
+end 
+
+function identity(inplanes::Int, planes::Int, stride::Int, expansion::Int)
+  inplanes = inplanes;
+  planes = planes;
+  downsample =  Chain(Conv((1, 1), inplanes=>planes * expansion, stride=stride),
+  BatchNorm(planes * expansion, λ=relu));
+  return identity
+end
+
+function resnet()
+  planes = 64;
+  inplanes = 64;
+  expansion = 1;
+  stride = 1;
+  layers = Chain(
+      Conv((7, 7), 3=>planes, stride=(2, 2), pad=(3, 3)),
+      BatchNorm(planes, λ=relu),
+      MaxPool((3, 3), stride=(2, 2), pad=(1, 1))
+      );
+    
+  push!(layers, SkipConnection(layers, basicblock(inplanes, planes, stride));
```

Comment:
I don't think I've explained `SkipConnection` well. Below is a screen grab from the ResNet paper (found [here](https://www.cv-foundation.org/openaccess/content_cvpr_2016/papers/He_Deep_Residual_Learning_CVPR_2016_paper.pdf)) that illustrates a ResNet architecture. I have marked up a red rectangle on the image. That rectangle is a single `SkipConnection`. As you can see, the ResNet architecture is many `SkipConnection`s in a row (except the start and end).

So, in your code, you should be building an array of layers, where each element is a `SkipConnection`. You do have the general right idea, but you are constructing `SkipConnection` wrong. Let's label the args of the constructor as `SkipConnection(A, B)`. In the image, you see a curved arrow (the skip) and a series of blocks between the curved arrow. `A` must be that series of blocks. You can see that the output of the series of blocks and the end of the curved arrow merge. `B` must be what operation is done to merge these outputs.

Therefore, your line here should look like
```julia
push!(layers, SkipConnection(basicblock(...), identity(...))
```

<img width="144" alt="Screen Shot 2020-06-25 at 9 01 13 AM" src="https://user-images.githubusercontent.com/4811350/85734371-9b155a80-b6c2-11ea-82d2-df1017779d6e.png">